### PR TITLE
Refactor Parametric solve for better performance and better use of Manopt.jl

### PR DESCRIPTION
**Pull request by Affie**
Progress tracked in https://github.com/JuliaRobotics/IncrementalInference.jl/issues/1546
Needs: https://github.com/JuliaRobotics/IncrementalInference.jl/pull/1776

### CODE REVIEW

File: test/testSphereMani.jl [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Manifolds: identity_element
 #FIXME REMOVE! this is type piracy and not a good idea, for testing only!!!
 Manifolds.identity_element(::Sphere{2, ℝ}) = SVector(1.0, 0.0, 0.0)
 Manifolds.identity_element(::Sphere{2, ℝ}, p::AbstractVector) = SVector(1.0, 0.0, 0.0) # Float64[1,0,0]
+DFG.getPointIdentity(::Sphere{2, ℝ}) = SVector(1.0, 0.0, 0.0)
```
Comment (dehann) [inline test/testSphereMani.jl:17]: perhaps just add a comment that this is an abuse of notation  (Sphere not being a group).  

I have been thinking about in future perhaps we should change this call to `getPointDefault` or `getPointFallback` and avoid use of identity on this context.  The reason for waiting on such a signature chance for me is first completing the solver upgrades to full Riemannian first and then come back.

File: src/parametric/services/ParametricUtils.jl [hunk 2]
```diff
@@ -294,34 +310,32 @@ function GraphSolveBuffers(@nospecialize(M), ::Type{T}) where {T}
...
-struct GraphSolveContainer
+struct GraphSolveContainer{CFT}
   M::AbstractManifold # ProductManifold or ProductGroup
   buffers::OrderedDict{DataType, GraphSolveBuffers}
   varTypes::Vector{DataType}
   varTypesIds::OrderedDict{DataType, Vector{Symbol}}
-  cfdict::OrderedDict{Symbol, CalcFactorMahalanobis}
   varOrderDict::OrderedDict{Symbol, Tuple{Int, Vararg{Int}}}
-  # cfarr::AbstractVector # TODO maybe <: AbstractVector(CalcFactorMahalanobis)
+  cfv::ArrayPartition{CalcFactorMahalanobis, CFT}
```
Comment (dehann) [inline src/parametric/services/ParametricUtils.jl:319]:
```suggestion
  cfv::ArrayPartition{<:CalcFactorMahalanobis, CFT}
```

Comment (Affie) [reply src/parametric/services/ParametricUtils.jl:319]: The ArrayPartition should be able to store a mixture of any CalcFactor type, so the `<:` will not work here.

Comment (Affie) [reply src/parametric/services/ParametricUtils.jl:319]: I'm not actually sure if the first parameter is used at all. 
I'm thinking of moving the optim.jl solve to the attic in any case.

**Pull request merged**